### PR TITLE
[fix] #4286 RFC Process Improvement and Documentation Update

### LINKED ISSUE #4286: [Docs] Add table of Contents in rfc/README.md
### Description

While going through the rfc/README.md file I felt that there should be a table of content for a better navigation of sub topics in the README.md

# Reproduction Steps
Make a table of contents for all the headings in the file and make them available by clicking on the link of the same.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

### PR DESCRIPTION
**Pull request by adarsh-jha-dev**
## Describe your changes

I've made improvements to the RFC Process section in the README, including restructuring and adding a table of contents for better readability. Additionally, I've made some minor grammar and formatting adjustments for clarity.

Fixes #4286 

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

## Screenshots

N/A

## Note to reviewers

No specific notes for reviewers.

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]
LGTM, thanks

**Comment by adarsh-jha-dev** [comment]
> LGTM, thanks

You're always welcome, will be pleased if merged successfully.

**Comment by adarsh-jha-dev** [comment]
@pingsutw , can you just let me know if there are further changes needed or you can merge this?

File: rfc/README.md [hunk 1]
```diff
@@ -2,105 +2,119 @@
...
 
-RFCs for vital features may need to be implemented immediately, whereas other RFCs may need to wait for the right developer to be available. 
-
-The author of an RFC may not always be the one to implement it, but if they do, they are welcome to submit an implementation for review after the RFC has been accepted. 
+RFCs for vital features may need to be implemented immediately, whereas other RFCs may need to wait for the right developer to be available. The author of an RFC may not always be the one to implement it, but if they do, they are welcome to submit an implementation for review after the RFC has been accepted. 
 
 ## RFC Postponement
 
-Rejected RFCs may be labelled as "postponed". This means the PR has been closed without implementation, because either:
+Rejected RFCs may be labeled as "postponed." This means the PR has been closed without implementation, because either:
```
Comment (davidmirror-ops) [inline rfc/README.md:103]: misplaced dot

**Review by davidmirror-ops — APPROVED** [review]
Thanks! Looks good.

**Comment by adarsh-jha-dev** [comment]
> Thanks! Looks good.
You're always welcome, will be pleased if merged successfully.

**Comment by samhita-alla** [comment]
@adarsh-jha-dev, can you create an issue for the same and link the PR to the issue?

**Comment by adarsh-jha-dev** [comment]
Hey @samhita-alla I have created  an issue #4286  for this PR and linked the same to this PR

**Comment by adarsh-jha-dev** [comment]
> @adarsh-jha-dev, can you create an issue for the same and link the PR to the issue?

Done, I request you to please proceed further.

**Pull request merged**